### ISSUE TITLE
Why detectron?

### ISSUE THREAD

**Issue by Noiredd** [trqminh/aistron#2]
Hi! I've been long thinking about creating something like this, one repository with all reference AIS methods with unified data pipeline and evaluation. Kudos to you for actually starting this project!

My question: why start from detectron? I don't mean to belittle this framework, but I personally find it difficult to grasp (perhaps overengineered?), and I'd risk saying that it's too abstract for research purposes. While it's my personal preference to work with pure PyTorch (with torchvision for RCNN backbone) as much as possible, I'd argue that a simpler setup would be objectively better for research (in short: all the code in one place -> easier access to minute details -> simpler debugging). In lieu of a proof, I reimplemented ORCNN in pure PyTorch the other day (with a ~~plan~~ hope to port other major AIS architectures as well) - code is private at least for now.

So my question is: do you see any significant advantages of using a highly complex and abstract framework such as detectron? I know many implementations already use it, making this a sort of "default" choice. But perhaps, looking forward, having a unified set of _simple_ implementations that are easy to modify and extend - could be a good thing? What do you think?

Your project is still in early stages, so maybe it's not too late to take such route ;) I for one would be happy to contribute to such an endeavor, if you were willing to cooperate :)

**Comment by trqminh** [comment]
Thank you for your interest and comments. It should be a long answer so I will edit this message and answer you as soon as as I can.

**Comment by zhending111** [comment]
good job！Are you still doing this？

**Comment by trqminh** [comment]
@zhending111 Yes, I am still doing this. I have been working on other things, I'll continue developing this during winter break. Btw, you're welcome to contribute!

**Comment by zhending111** [comment]
> @zhending111 Yes, I am still doing this. I have been working on other things, I'll continue developing this during winter break. Btw, you're welcome to contribute!

Sure！I am researching the replacement of the baseline with a regular Mask R-CNN

**Comment by Glaciercoool** [comment]
really good job！！！
